packages/flutter/lib/src/painting/continuous_stadium_border.dart

```diff
@@ -0,0 +1,278 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'basic_types.dart';
+import 'borders.dart';
+import 'edge_insets.dart';
+
+/// Creates a continuous stadium shape.
+///
+/// A shape similar to a stadium, but with a smoother transition from
+/// each linear edge to its 180º curves.
+///
+/// In this shape, the curvature of each 180º curve over the arc is
+/// approximately a gaussian curve instead of a step function as with a
+/// traditional half circle round.
+///
+/// In the event that the height or width of the bounding rectangle is less than
+/// 2x its radius, the curve radius will become smaller to keep the shape from
+/// becoming a lozenge.
+///
```

Comment:
describe somewhere where the axis of the 2 half circles is going to be respective to the height/width ratio.